core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerServiceSpec.js

```diff
@@ -0,0 +1,344 @@
+      expect(explorationStatesService.getState(testStateName)
+        .interaction.answer_groups).toContain(expectedNewAnswerGroups);
+    });
+  */
+
+  it('should save the default outcome', function() {
+    var testStateName = 'Introduction';
+    var expectedNewDefaultOutcome = {
+      dest: 'Introduction',
+      feedback: [
+        '<p>One is wrong bcz ofisEmpty many reasons</p>'
+      ]
+    };
+    simpleEditorManagerService.tryToInit();
+    simpleEditorManagerService.saveDefaultOutcome(
+      testStateName, expectedNewDefaultOutcome);
+    expect(explorationStatesService.getState(testStateName)
+      .interaction.default_outcome).toEqual(expectedNewDefaultOutcome);
+  });
+
+  /**
+    * Not working, needs debugging.
+  it('should save the bridge html', function () {
+    var testStateName = 'Introduction';
+    var expectedNewBridgeHtml = '<p> lets move to next Qsn </p>';
+    var expectedSimpleEditorManagerQuestionList = questionList;
+    simpleEditorManagerService.tryToInit()
+    simpleEditorManagerService.saveBridgeHtml(
+      testStateName, expectedNewBridgeHtml);
+    expect(explorationStatesService
+      .getState(expectedSimpleEditorManagerQuestionList
+        .getNextStateName(testStateName)).content)
+          .toEqual(expectedNewBridgeHtml);
+  });
+*/
+  it('should add new question and new state', function() {
+    var expectedSimpleEditorQuestionList = questionList;
+    var expectedLastStateName;
+    var expectedDefaultOutcome;
+    var DEFAULT_INTERACTION = {
+      ID: 'MultipleChoiceInput',
+      CUSTOMIZATION_ARGS: {
+        choices: {
+          value: ['<p>Option 1</p>']
+        }
+      }
+    };
+
+    simpleEditorManagerService.tryToInit();
+    expectedLastStateName =
+      expectedSimpleEditorQuestionList.getLastQuestion()
+        .getDestinationStateName();
+
+    expectedDefaultOutcome = {
+      dest: expectedLastStateName,
+      feedback: [''],
+      param_changes: []
+    };
+
+    simpleEditorManagerService.addNewQuestion();
+    expect(explorationStatesService.getState(expectedLastStateName)
+      .interaction.id).toEqual(DEFAULT_INTERACTION.ID);
+    expect(explorationStatesService.getState(expectedLastStateName)
+      .interaction.customization_args)
+        .toEqual(DEFAULT_INTERACTION.CUSTOMIZATION_ARGS);
+    expect(explorationStatesService.getState(expectedLastStateName)
+      .interaction.default_outcome).toEqual(expectedDefaultOutcome);
+    var expectedStateData =
+      simpleEditorShimService.getState(expectedLastStateName);
+    expect(JSON.stringify(simpleEditorManagerService.getQuestionList()))
+      .toContain(JSON.stringify(questionObjectFactory.create(
+        expectedLastStateName, expectedStateData.interaction, '')));
+
+    // It should also add the new state.
+    simpleEditorManagerService.addState();
+    var expectedAllStates = explorationStatesService.getStates();
+    expect(expectedAllStates['Question 2']).toBeDefined();
```

Comment:
I have defined a separate test for it and that checks for, if the newly added state has all the correct properties.